fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/api/dimension/v1/FabricDimensionType.java

```diff
@@ -0,0 +1,246 @@
+	 *
+	 *   public static final FabricDimensionType MY_DIMENSION
+	 *       = FabricDimensionType.builder()
+	 *           .defaultPlacement((oldEntity, destination, portalDir, horizontalOffset, verticalOffset) ->
+	 *           		new BlockPattern.TeleportTarget(new Vec3d(0, 100, 0), teleported.getVelocity(), 0))
+	 *           .factory(MyDimension::new)
+	 *           .skyLight(true)
+	 *           .buildAndRegister();}</pre>
+	 *
+	 * <p>Builder instances can be reused; it is safe to call {@link #buildAndRegister(Identifier)} multiple
+	 * times (with different identifiers) to build and register multiple dimension types in series.
+	 * Each new dimension type uses the settings of the builder at the time it is built.
+	 *
+	 * @see FabricDimensionType#builder()
+	 */
+	public static final class Builder {
+		private EntityPlacer defaultPlacer;
+		private BiFunction<World, DimensionType, ? extends Dimension> factory;
+		private int desiredRawId = 0;
+		private boolean skyLight = true;
+		private class_4545 biomeFunction = class_4547.INSTANCE;
+
+		private Builder() {
+		}
+
+		/**
+		 * Set the default placer used when teleporting entities to dimensions of the built type.
+		 * The default placer must be set before building a dimension type.
+		 *
+		 * <p> A dimension type's default placer must never return {@code null} when its
+		 * {@link EntityPlacer#placeEntity(Entity, ServerWorld, Direction, double, double) placeEntity} method
+		 * is called.
+		 *
+		 * @param defaultPlacer a default entity placer for dimensions of the built type
+		 * @return this {@code Builder} object
+		 * @throws NullPointerException if {@code defaultPlacer} is {@code null}
+		 */
+		public Builder defaultPlacer(EntityPlacer defaultPlacer) {
+			Preconditions.checkNotNull(defaultPlacer);
+
+			this.defaultPlacer = defaultPlacer;
+			return this;
+		}
+
+		/**
+		 * Set the factory used to create new {@link Dimension} instances of the built type.
+		 * The dimension factory must be set before building a dimension type.
+		 *
+		 * @param factory a function creating new {@code Dimension} instances
+		 * @return this {@code Builder} object
+		 * @throws NullPointerException if {@code factory} is {@code null}
+		 */
+		public Builder factory(BiFunction<World, DimensionType, ? extends Dimension> factory) {
+			Preconditions.checkNotNull(factory);
+
+			this.factory = factory;
+			return this;
+		}
+
+		/**
+		 * Set whether built dimension types use skylight like the Overworld.
+		 * If this method is not called, the value defaults to {@code true}.
+		 *
+		 * @param skyLight {@code true} if the dimension of the built type should use skylight,
+		 *                 {@code false} otherwise
+		 * @return this {@code Builder} object
+		 */
+		public Builder skyLight(boolean skyLight) {
+			this.skyLight = skyLight;
+			return this;
+		}
+		
+		/**
+		 * Set biome function associated with the dimension.
```

Comment:
This sets the way a biome should be obtained from the updated biome array storage with a seed, the biome storage, and block x,y,z. It is not only for generation but also on the call path when game mechanism checks biomes.